### PR TITLE
bpo-40823: Use loadTestsFromTestCase() iso. makeSuite() in sqlite3 tests

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue40823

### CODE REVIEW

**Review by berkerpeksag — CHANGES_REQUESTED** [review]
I don't have a strong objection to this, but if we are going to drop `Check` prefix, I'd prefer using `test_*` instead of `test*`.

**Comment by erlend-aasland** [comment]
> I don't have a strong objection to this, but if we are going to drop `Check` prefix, I'd prefer using `test_*` instead of `test*`.

No problem. I'll rebase onto master and fix the naming.

File: Lib/sqlite3/test/types.py [hunk 1]
```diff
@@ -313,19 +313,19 @@ def tearDown(self):
...
         self.con.close()
 
-    def CheckCursorDescriptionCTESimple(self):
+    def test_cursor_description_cte_simple(self):
         self.cur.execute("with one as (select 1) select * from one")
         self.assertIsNotNone(self.cur.description)
         self.assertEqual(self.cur.description[0][0], "1")
 
-    def CheckCursorDescriptionCTESMultipleColumns(self):
+    def test_cursor_description_ctes_multiple_columns(self):
```
Comment (erlend-aasland) [inline Lib/sqlite3/test/types.py]:
```suggestion
    def test_cursor_description_cte_multiple_columns(self):
```

@berkerpeksag: ctes => cte?

**Comment by erlend-aasland** [comment]
PTAL, @berkerpeksag. Rebased onto master with snake case method naming. IMHO, it looks really nice now.

**Comment by erlend-aasland** [comment]
I have made the requested changes; please review again

File: Lib/sqlite3/test/types.py (hunk 1 above)
Comment (berkerpeksag) [reply Lib/sqlite3/test/types.py]: CTE sounds good to me.

File: Lib/sqlite3/test/userfunctions.py [hunk 2]
```diff
@@ -494,17 +494,18 @@ def authorizer_cb(action, arg1, arg2, dbname, source):
...
-            authorizer_suite,
-            unittest.makeSuite(AuthorizerRaiseExceptionTests),
-            unittest.makeSuite(AuthorizerIllegalTypeTests),
-            unittest.makeSuite(AuthorizerLargeIntegerTests),
-        ))
+    tests = [
+        AggregateTests,
+        AuthorizerTests,
+        FunctionTests,
+
```
Comment (berkerpeksag) [inline Lib/sqlite3/test/userfunctions.py]: Nitpick: I'd drop the extra newline here.

File: Misc/NEWS.d/next/Tests/2020-05-30-13-39-22.bpo-40823.yB7K5w.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Use :meth:`unittest.TestLoader().loadTestsFromTestCase` iso.
```
Comment (berkerpeksag) [inline Misc/NEWS.d/next/Tests/2020-05-30-13-39-22.bpo-40823.yB7K5w.rst]: Nitpick: I'd prefer using the long form of iso here.

File: Misc/NEWS.d/next/Tests/2020-05-30-13-39-22.bpo-40823.yB7K5w.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
...
+:meth:`unittest.makeSuite` in :mod:`sqlite3` tests.
```
Comment (berkerpeksag) [inline Misc/NEWS.d/next/Tests/2020-05-30-13-39-22.bpo-40823.yB7K5w.rst]: Nitpick: Please add "Patch by Your Name."

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

**Comment by erlend-aasland** [comment]
I have made the requested changes; please review again

**Comment by erlend-aasland** [comment]
GitHub messed up the encoding again! See commit fd4f651. I must remember no to use the suggestion feature for these files.  We should normalise the encoding:

```
$ file Lib/sqlite3/test/*.py
Lib/sqlite3/test/__init__.py:      empty
Lib/sqlite3/test/backup.py:        Python script text executable, ASCII text
Lib/sqlite3/test/dbapi.py:         Python script text executable, ISO-8859 text
Lib/sqlite3/test/dump.py:          Python script text executable, ASCII text
Lib/sqlite3/test/factory.py:       Python script text executable, ISO-8859 text
Lib/sqlite3/test/hooks.py:         Python script text executable, ISO-8859 text
Lib/sqlite3/test/regression.py:    Python script text executable, ISO-8859 text
Lib/sqlite3/test/transactions.py:  Python script text executable, ISO-8859 text
Lib/sqlite3/test/types.py:         Python script text executable, UTF-8 Unicode text
Lib/sqlite3/test/userfunctions.py: Python script text executable, UTF-8 Unicode text
```

**Comment by erlend-aasland** [comment]
We converted Lib/sqlite3/test/userfunctions.py to UTF-8 in #20448. I'd say we fix the rest right away. Agreed, @berkerpeksag?

**Comment by berkerpeksag** [comment]
> The same thing happened in #20448, and we ended up just converting Lib/sqlite3/test/userfunctions.py to UTF-8 in the same PR. I'd say we convert the rest right away. Agreed, @berkerpeksag?

I failed to see what's wrong here and #20448 has lots of comments. Why do you want to convert encodings of these files?

**Comment by erlend-aasland** [comment]
> I failed to see what's wrong here and #20448 has lots of comments. Why do you want to convert encodings of these files?

Commit fd4f651 was auto-generated by GitHub from the "suggestion" I set up in https://github.com/python/cpython/pull/20538#discussion_r552926126. As you notice from the diff, GitHub silently converted the file from ISO-8859 encoding to UTF-8 encoding (ignoring the `#-*- coding: iso-8859-1 -*-` header), in addition to applying the "suggestion" I set up. Over at #20448 the same thing happened; Victor Stinner set up a "suggestion", which I applied, and then GitHub silently converted Lib/sqlite3/test/userfunctions.py from ISO-8859 to UTF-8. Instead of reverting the encoding change GitHub made, we decided to remove the `#-*- coding: iso-8859-1 -*-` header instead, since GitHub obviously dislikes ISO-8859.

**Comment by berkerpeksag** [comment]
Ah, I see now, thanks for the explanation! Personally, I prefer reverting the commit generated by GitHub and apply the suggested change manually. I don't like when companies try to impose their own practices (it doesn't mean they are bad practices) on a perfectly fine piece of code :)

**Comment by erlend-aasland** [comment]
… or we can revert the encoding change and create an issue for normalising the encodings.

> Ah, I see now, thanks for the explanation! Personally, I prefer reverting the commit generated by GitHub and apply the suggested change manually. I don't like when companies try to impose their own practices (it doesn't mean they are bad practices) on a perfectly fine piece of code :)

No problem. I'll open an issue for normalising the file encodings. Consistence FTW :)

**Comment by erlend-aasland** [comment]
Reverted in f93a63b1cd6cb9c9dd6ddbe6700fc1ce3f84862a. PTAL, @berkerpeksag

**Review by berkerpeksag — APPROVED** [review]
LGTM!

**Pull request merged**